File: CrmApp.Client/src/models/DTO/ValueNameDTO.ts
export class ValueNameDTO<T=number> {

    constructor(_val: T, _name: string, _icon?: string, _iconClass?: string, _key?: string, _isDefault?: boolean) {
        this.value = _val;
        this.name = _name;
        this.icon = _icon;
        this.iconClass = _iconClass;
        this.key = _key;
        this.isDefault = _isDefault;
    }

    value: T;
    name: string;
    icon?: string;
    iconClass?: string;
    key?: string;
    isDefault?: boolean;
}

File: CrmApp.Domain/DTO/ValueNameDTO.cs
using CrmApp.Domain.Tools;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CrmApp.Domain.DTO
{
    public class ValueNameDTO<T>
    {
        public ValueNameDTO(T value, string name, string? icon = null, string? iconClass = null, string? key = null) 
        { 
            this.Value = value;
            this.Name = name;
            this.Icon = icon;
            this.IconClass = iconClass;
            this.Key = key;
        }

        public T Value { get; set; }
        public string Name { get; set; }
        public string? Icon { get; set; }
        public string? IconClass { get; set; }
        public string? Key { get; set; }
    }
}
